src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -144,9 +144,12 @@ private async Task HandleClusterDatabaseChanged(string databaseName, long index,
                             }
                             else
                             {
-                                using (ShardedDatabasesCache.RemoveLockAndReturn(databaseName, (databaseContext) => databaseContext.Dispose(), out var _))
+                                using (ShardedDatabasesCache.RemoveLockAndReturn(databaseName, (databaseContext) => databaseContext.Dispose(), out _))
                                 {
                                 }
+
+                                context.Transaction.InnerTransaction.LowLevelTransaction.OnDispose += 
```

Comment:
check if tx is commited

Reply:
removed it from the `OnDispose`, calling `DeleteStorageFor` directly